lib/kimono_labs_client/web_client.rb

```diff
@@ -0,0 +1,39 @@
+module KimonoLabsClient
+  # Web client for fetching a json endpoint's content via http.
+  class WebClient
+    attr_reader :api_key
+
+    # initializer - named params are required and throw ArgumentError automatically
+    # https://robots.thoughtbot.com/ruby-2-keyword-arguments
+    def initialize(api_key:)
+      @api_key = api_key
+      @http = http
```

Comment:
Instead of setting this up in the initializer, what if you had the `http` method cache the faraday object itself? I can store the object in `@http` still, but only initialize `Faraday` if one doesn't exist. That way you can just use the `http` method directly wherever you need it without additional setup _and_ it is just in time evaluation.